[Feat] PlaceViewCollectionViewCell 레이아웃 세팅

jaminleee opened the pull request:
## 🔥*Pull requests*

⛳️ **작업한 브랜치**
- #22
👷 **작업한 내용**
<!-- 작업한 내용을 적어주세요. -->
- PlaceViewCollectionViewCell들 레이아웃 설정

## 🚨 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
- PlaceView 첫번째 셀의 그림자를 설정하면 radius가 안먹고, radius하면 그림자 안먹어서 일단 놔뒀습니다

## 📸 스크린샷
|기능|스크린샷|
|:--:|:--:|
|GIF|<img src = "https://github.com/SOPT-33RD-APP-NAVERPAY/NaverPay-iOS/assets/91969458/bca58939-2112-4a27-a8c8-c1e1f24e1bdc" width ="250">|

## 📟 관련 이슈
- Resolved: #22

jaminleee committed:
[Feat] PlaceView 이미지 파일 추가

jaminleee committed:
[Feat] nearPlaceCardImageView 레이아웃 세팅

jaminleee committed:
[Feat] RecommendCollectionViewCell 레이아웃 세팅

jaminleee committed:
[Feat] BrandCollectionViewCell 레이아웃 세팅

jaminleee committed:
[Feat] PlaceView 이미지 파일 추가

jaminleee committed:
[Feat] ImageLiterals 수정 (#8)

jaminleee committed:
[Feat] RecommendCollectionViewCell 레이아웃 세팅

jaminleee committed:
[Feat] PlaceViewController 헤더뷰 값 설정

jaminleee committed:
[Feat] PlaceViewController 헤더 뷰 설정 (#22)

kim-seonwoo commented on NaverPay/Scenes/Place/Cells/BrandCollectionViewCell.swift:
```diff
@@ -10,6 +10,21 @@ import UIKit
+        return imageView
+    }()
+    
+    private let brandNameLabel:UILabel = {
```
NPLabel도입 해보는 것도 좋을 것 같아요~!

kim-seonwoo reviewed (commented):
UI 구현 수고하셨습니다~~! 파이팅 입니다~~😎

jaminleee committed:
[Feat] 국내 | 해외 header로 추가 (#22)

jaminleee committed:
[Feat] 국내/해외 cell 생성 및 추가

sjk4618 commented on NaverPay/Scenes/Place/Cells/BrandCollectionViewCell.swift:
```diff
@@ -10,6 +10,21 @@ import UIKit
 class BrandCollectionViewCell: UICollectionViewCell {
```
final!

sjk4618 commented on NaverPay/Scenes/Place/Cells/BrandCollectionViewCell.swift:
```diff
@@ -10,6 +10,21 @@ import UIKit
 class BrandCollectionViewCell: UICollectionViewCell {
     static let identifier:String = "BrandCollectionViewCell"
     
+    private let brandCardImageView:UIImageView = {
```
: 쓰고 한칸 띄어주기!

sjk4618 commented on NaverPay/Scenes/Place/Cells/BrandCollectionViewCell.swift:
```diff
@@ -21,20 +36,26 @@ class BrandCollectionViewCell: UICollectionViewCell {
+        self.layer.borderColor = UIColor.gray.cgColor
+        
+        [brandCardImageView, brandNameLabel].forEach{
             self.contentView.addSubview($0)
```
addsubviews extension 사용하면 좋을 것 같네요~

sjk4618 commented on NaverPay/Scenes/Place/Cells/BrandCollectionViewCell.swift:
```diff
@@ -21,20 +36,26 @@ class BrandCollectionViewCell: UICollectionViewCell {
+        brandNameLabel.snp.makeConstraints{
+            $0.top.equalTo(brandCardImageView.snp.bottom).offset(14)
+            $0.centerX.equalToSuperview()
+            $0.bottom.equalToSuperview().inset(10)
```
라벨의 경우 top을 잡아주고 centerX로 위치를 잡아주면 바텀을 잡아주지않아도 저절로 라벨의 font값따라 정해집니다!

sjk4618 commented on NaverPay/Scenes/Place/ViewControllers/PlaceViewController.swift:
```diff
@@ -86,27 +86,40 @@ extension PlaceViewController: UICollectionViewDataSource {
                     return TitleCollectionReusableView()
                 }
-                header.label.text = "\(userName)님과 가까운 장소"
+                header.headerLabel.text = "\(userName)님과 가까운 장소"
+                header.headerButton.setImage(ImageLiterals.PlaceView.placeButtonmapIcon, for: .normal)
+                header.headerButton.backgroundColor = .main_green
+                header.headerButton.setTitle("지도로 보기", for: .normal)
```
이 부분을 데이터를 넘겨서 해주면 좋을 것 같습니다~~ 
예시로는 제 코드를 보면 didset 부분을 보시면 됩니당

sjk4618 reviewed (commented):
수고하셨습니다!!
: 뒤에 한칸 띄어쓰기랑 addsubviews extension을 사용하면 좋을 것 같네요~~
header나 cell에 데이터 넘겨주는 것도 didset이나 메서드로 넘겨주면 좋을 것 같아요~

jaminleee committed:
[Chore] 코드 리뷰 반영 (#22)

jaminleee committed:
[Fix] conflict 해결 (#22)

# Conflicts:
#	NaverPay.xcodeproj/project.pbxproj
#	NaverPay/Scenes/Place/ViewControllers/PlaceViewController.swift
